### ISSUE TITLE
Verify code fail on Blockscout but success on Etherscan

### ISSUE THREAD

**Issue by guagualvcha** [poanetwork/blockscout#3191]
The contract deployed on ropsten network:
https://ropsten.etherscan.io/address/0x1f935926e8e39cff3fb84912b7fec986ce1ed7e0#code

The contract deployed on POA Sokol network:
https://blockscout.com/poa/sokol/address/0x8BCE6429A197A088C1457Fd3b757cF0732629Eaf/transactions

How I deploy the contract:
![image](https://user-images.githubusercontent.com/7310198/87281679-0d40c880-c526-11ea-9732-6337f1e18a98.png)
And I try publish and verify the contract:
![image](https://user-images.githubusercontent.com/7310198/87283660-17fc5d00-c528-11ea-841b-2a2b31c39ec5.png)

The error message is:
![image](https://user-images.githubusercontent.com/7310198/87282025-69a3e800-c526-11ea-885d-464b44bf3f1f.png)

The log we can find on our blocksout is that:
![image](https://user-images.githubusercontent.com/7310198/87282153-8dffc480-c526-11ea-943b-33b34f7f0f6d.png)

Because the code: 
```
  def evaluate_authenticity(address_hash, params) do
    latest_evm_version = List.last(CodeCompiler.allowed_evm_versions())
    evm_version = Map.get(params, "evm_version", latest_evm_version)

    Enum.reduce([evm_version | previous_evm_versions(evm_version)], false, fn version, acc ->
      case acc do
        {:ok, _} = result ->
          result

        _ ->
          cur_params = Map.put(params, "evm_version", version)
          verify(address_hash, cur_params)
      end
    end)
  end
```
will try with three evm version: Istanbul,constantinople,petersburg.

The first try with Istanbul, the compile is success, but bytecode mismatch, I don't know why it success on Ether-scan but failed on Blocksout?

**Comment by guagualvcha** [comment]
Here is the flatten contract, many thanks if you guys can help and verify this case. 
[Comp.txt](https://github.com/poanetwork/blockscout/files/4911265/Comp.txt)

**Comment by vbaranov** [comment]
Seems like `ABIEncoderV2` causes verification issue. Related issue https://github.com/poanetwork/blockscout/issues/2392#issuecomment-601587610

**Comment by guagualvcha** [comment]
@vbaranov thanks for your information, but how about contracts compiled with Solc 0.5.x with ABIEncoderV2? Any way that I can work around it?  Can I change any config of Blockscout?

**Comment by vbaranov** [comment]
@guagualvcha it is an unresolved contracts verification issue for 0.5.x compilers yet. Indeed, compiled bytecode and bytecode, stored in the blockchain, mismatch.

**Comment by guagualvcha** [comment]
Do you have any plan to solve this?

**Comment by igorbarinov** [comment]
> Do you have any plan to solve this?

As Victor mentioned, there is a problem with compiler versions `0.5.x` to match the code from the blockchain and compiled by BlockScout. Etherscan is not opensource so we can't check with them.

Can you compile contracts locally to match the bytecode? Would be great to know your compiler parameters if you can

**Comment by vbaranov** [comment]
@guagualvcha I found a root cause of the issue. When experimental features are enabled an additional hex from `experimental` is added and we need to account it when dividing contract bytecode from swarm/ipfs hash of the contract metadata. I've prepared a fix for verification of contracts with experimental features enabled https://github.com/poanetwork/blockscout/pull/3202. I also installed to Sokol instance of Blockscout. Could you confirm that the fix does resolve issue with verification? I successfully verified similar contract https://blockscout.com/poa/sokol/address/0xA5615Bc61B45e1778E20a5ccF78943d1Dd50FFbe/contracts with ABIEncoderV2 enabled

**Comment by guagualvcha** [comment]
sure. I will have a try and feedback here.

**Comment by vbaranov** [comment]
@guagualvcha the fix should work for all versions >= 0.5.16. A verified contract which I posted compiled with 0.5.17.
